app/news-page/stores/news.store.ts

```diff
@@ -0,0 +1,127 @@
+import { action, computed, observable, runInAction } from 'mobx';
+
+import { inject, injectable } from '@servicetitan/react-ioc';
+
+import { User } from '../../common/models/user';
+import { AuthStore } from '../../common/stores/auth.store';
+import { UserApi } from '../../users-page/api/user.api';
+import { NewsApi } from '../api/news.api';
+import { Post } from '../models/post';
+import { EditPostStore } from './edit-post.store';
+
+export type PostCardData = [Post, User['login']];
+
+@injectable()
+export class NewsStore {
+    @observable
+    editPostStore?: EditPostStore;
+
+    constructor(
+        @inject(UserApi) private userApi: UserApi,
+        @inject(NewsApi) private newsApi: NewsApi,
+        @inject(AuthStore) private authStore: AuthStore,
+    ) {
+    }
+
+    @observable
+    serverErrors: string[] = [];
+
+    @observable
+    private users?: User[];
+
+    @observable
+    private news?: Post[];
+
+    @computed
+    get postCardData(): PostCardData[] | undefined {
+        if (!this.users || !this.news) {
+            return void 0;
+        }
+
+        const getAuthorName = getUserName(this.users);
+        return (
+            this.news
+                .reverse()
+                .map<PostCardData>(p => [p, getAuthorName(p.authorId)])
+        );
+    }
+
+    @action
+    update = async () => {
+        this.serverErrors = [];
+
+        await Promise.all([
+            this.updateUsers(),
+            this.updateNews(),
+        ]);
+    };
+
+    @action
+    private async updateUsers() {
+        this.users = void 0;
+        const r = await this.userApi.getAll();
+        if (r.status !== 200) {
+            runInAction(() => this.serverErrors.push(r.statusText));
+        }
+
+        runInAction(() => this.users = r.data);
+    }
+
+    @action
+    private async updateNews() {
+        this.news = void 0;
+        const r = await this.newsApi.getAll();
+        if (r.status !== 200) {
+            runInAction(() => this.serverErrors.push(r.statusText));
+        }
+
+        runInAction(() => this.news = r.data);
+    }
+
+    onCreate = () => this.onEdit({ authorId: this.authStore.getUser()!.id });
```

Comment:
I'm not sure that the client should send the author, it wouldn't be secure.

Reply:
It looks like we need a different model for the create/update.

Reply:
> it wouldn't be secure

We have not security at all.
Perhaps we should add some (node.js?) server to original repo, so that the new employee does not implement it himself.